programs/mango-v4/src/instructions/perp_consume_events.rs

```diff
@@ -44,6 +44,41 @@ macro_rules! load_mango_account {
     };
 }
 
+#[derive(AnchorSerialize, AnchorDeserialize)]
+pub enum FillCategory {
+    OpenShort,
+    OpenLong,
+    IncreaseShort,
+    IncreaseLong,
+    DecreaseShort,
+    DecreaseLong,
+}
+
+fn get_fill_category(pre_fill_base_position: i64, side: Side) -> FillCategory {
+    // Note: if base position is 100 and 200 is sold (final base position -100)
+    // this is defined as DecreaseLong (conceivably this could also be defined as OpenShort)
```

Comment:
I'm highly sceptical of the usefulness of this classification.
- Separating out OpenLong/Short will make people try to filter for that, but like this comment suggests: it won't work when the user's position crosses 0. So you cannot rely on the open events.
- And because you can go from -100 to +100 in a DecreaseShort, you can't rely on Increase events either

So I expect this will lead to queries that don't really work.

How about just logging the post lots for maker and taker? Then you can easily query for fills that increase a long with `post_taker_lots > 0 && taker_side == bid` etc. It will make the queries slightly more complex, but they'd be more likely to do the right thing imo.

Reply:
maybe we can split when it crosses into 2 events

Reply:
Thanks for the review.
Trying to get the best of both worlds here - I added some more enum states to cover all combinations of pre and post lots and bid/ask. Then also added in logging of the `post_taker_lots` and `post_maker_lots`.
My reasons is that the enum is helpful for convenience to someone querying the fills - but they can break it down further if they like by looking at the post lots. I can remove the enum though if you are not convinced.